packages/delayedInbox-l2msg/scripts/normalTx.js

```diff
@@ -1,152 +1,172 @@
-        from: l1Wallet.address
-    };
-
-    const gasLimit1 = await l1Provider.estimateGas(transactionl1Request)
-
-    transactionl1Request.gasLimit = gasLimit1
-
-    let resultsL1 = await l1Wallet.sendTransaction(transactionl1Request)
-
-
-    
-    const inboxRec = await resultsL1.wait()
-
+  /**
+   * We need to check if the sender has enough funds on l2 to pay the gas fee
+   */
+  if (l2Balance.lt(l2GasPrice.mul(l2GasLimit))) {
     console.log(
-        `Greeting txn confirmed on L1! 🙌 ${inboxRec.transactionHash}`
+      'You l2 balance is not enough to pay the gas fee, please bridge some ethers to l2.'
     )
-
-    const l1TxReceipt = new L1TransactionReceipt(inboxRec)
-
-    /**
-     * In principle, a single L1 txn can trigger any number of L1-to-L2 messages (each with its own sequencer number).
-     * In this case, we know our txn triggered only one
-     * Here, We check if our L1 to L2 message is redeemed on L2
-     */
-    const message = await l1TxReceipt.getL1ToL2Message(l2Wallet)
-    const status = await message.waitForStatus()
-    console.log(status)
-    if (status === L1ToL2MessageStatus.REDEEMED) {
-        console.log(`L2 txn executed 🥳 ${message.l2TxHash}`)
-    } else {
-        console.log(
-        `L2 retryable txn failed with status ${L1ToL2MessageStatus[status]}`
-        )
-    }
-
-    /**
-     * Note that during L2 execution, a retryable's sender address is transformed to its L2 alias.
-     * Thus, when GreeterL2 checks that the message came from the L1, we check that the sender is this L2 Alias.
-     * See setGreeting in GreeterL2.sol for this check.
-     */
-
-    /**
-     * Now when we call greet again, we should see our new string on L2!
-     */
-    const newGreetingL2 = await l2Greeter.greet()
-    console.log(`Updated L2 greeting: "${newGreetingL2}"`)
-    console.log('✌️')
+    return
+  }
+
+  /**
+   * We need extract l2's tx hash first so we can check if this tx executed on l2 later.
+   */
+  const l2SignedTx = await l2Wallet.signTransaction(transactionl2Request)
+
+  const l2Txhash = ethers.utils.parseTransaction(l2SignedTx).hash
+
+  /**
+   * Pack the message data to parse to delayed inbox
+   */
+  const sendData = ethers.utils.solidityPack(
+    ['uint8', 'bytes'],
+    [ethers.utils.hexlify(L2MSG_signedTx), l2SignedTx]
+  )
+  console.log('Now we get the send data: ' + sendData)
+
+  /**
+   * Process the l1 delayed inbox tx, to process it, we need to have delayed inbox's abi and use it to encode the
+   * function call data. After that, we send this tx directly to delayed inbox.
+   */
+  const ABI = [
+    'function sendL2Message(bytes calldata messageData) external returns(uint256)',
+  ]
+  const iface = new ethers.utils.Interface(ABI)
+  const calldatal1 = iface.encodeFunctionData('sendL2Message', [sendData])
+  const l1GasPrice = await l1Provider.getGasPrice()
+
+  const transactionl1Request = {
+    data: calldatal1,
+    to: process.env.INBOX_ADDR,
```

Comment:
lets get the inbox from the SDK using getNetwork instead of hardcoding in the env variable